client/dali_grpc_client.py

```diff
@@ -120,44 +127,36 @@ def main():
 
     print("Loading images")
 
-    image_data = load_images(FLAGS.img_dir if FLAGS.img_dir is not None else FLAGS.img)
+    image_data = load_images(FLAGS.img_dir if FLAGS.img_dir is not None else FLAGS.img,
+                             max_images=FLAGS.batch_size * FLAGS.n_iter)
+
     image_data = array_from_list(image_data)
+    print("Images loaded")
 
-    print("Images loaded, inferring")
+    for batch in tqdm(utils.batcher(image_data, FLAGS.batch_size, n_iterations=FLAGS.n_iter),
+                      desc="Inferring", total=FLAGS.n_iter):
```

Comment:
> How it behaves if total=-1?

Doesn't show progress bar

> In this case where the batch_size is integer, you can calculate it as len(image_data) // batch_size, can't you?

This would assume, that user wants to pass whole dataset. But he might want to have like 1000 images in the `image_data` and pass only 2 batches of 32 samples each